src/pudl/extract/xbrl.py

```diff
@@ -116,23 +110,24 @@ def convert_form(
     """
     datapackage_path = str(output_path / f"ferc{form.value}_xbrl_datapackage.json")
     metadata_path = str(output_path / f"ferc{form.value}_xbrl_taxonomy_metadata.json")
+
+    taxonomy_archive = datastore.get_taxonomy(form)
     # Process XBRL filings for each year requested
-    for year in form_settings.years:
-        taxonomy_archive, taxonomy_entry_point = datastore.get_taxonomy(year, form)
-        filings_archive = datastore.get_filings(year, form)
-        # if we set clobber=True, clobbers on *every* call to run_main;
-        # we already delete the existing base on `clobber=True` in `xbrl2sqlite`
-        run_main(
-            instance_path=filings_archive,
-            sql_path=sql_path,
-            clobber=False,
-            taxonomy=taxonomy_archive,
-            entry_point=taxonomy_entry_point,
-            form_number=form.value,
-            metadata_path=metadata_path,
-            datapackage_path=datapackage_path,
-            workers=workers,
-            batch_size=batch_size,
-            loglevel="INFO",
-            logfile=None,
-        )
+    filings_archives = [
+        datastore.get_filings(year, form) for year in form_settings.years
+    ]
+    # if we set clobber=True, clobbers on *every* call to run_main;
+    # we already delete the existing base on `clobber=True` in `xbrl2sqlite`
+    run_main(
+        filings=filings_archives,
```

Comment:
Because all taxonomies are bundled together, the extractor can now just take all of the filings at once.

Reply:
The main change int he API here seems to be that since all the taxonomies are included, the extractor can dynamically choose which taxonomy to use based on what year of data is being extracted internally?

Reply:
Yeah now the archiver will look inside each filing and save metadata mapping the filing to the taxonomy it used, then the extractor will use that metadata to decide which taxonomy to use when parsing.